### ISSUE TITLE
RichTextLabel lists support

### ISSUE THREAD

**Issue by ganlub** [godotengine/godot#21285]
**Godot version:**

v3.0.6

**OS/device including version:**

Archlinux

**Issue description:**

I wanted to display a list inside a RichTextLabel but it's not working properly.

Following the official bbcode [reference](https://bbcode.org/examples/?id=12), this is the output I'm getting:
![image](https://user-images.githubusercontent.com/1272617/44449801-6f6d4900-a5ef-11e8-8b84-9d206294e4c3.png)

On the one hand, there seems to be a parsing issue when using `[ol][/ol]` and `[ul][/ul]`.
On the other hand, documentation is not clear enough about the exact supported tags. [This](http://docs.godotengine.org/en/latest/tutorials/gui/bbcode_in_richtextlabel.html#reference) is the only reference I could find and [here](https://godot.readthedocs.io/en/3.0/classes/class_richtextlabel.html#class-richtextlabel-push-list) there's a note about `push_list` method being not fully implemented.

**Steps to reproduce:**
Add a RichTextLabel to your scene, enable bbcode and set this text to it:
```
Ordered list:
[ol]
[li]Item one[/li]
[li]Item two[/li]
[/ol]
Unordered list:
[ul]
[li]Item one[/li]
[li]Item two[/li]
[/ul]
Another variant:
[list]
[li]Item one[/li]
[li]Item two[/li]
[/list]
Other ways of indicating list items:
[list]
[*]Here's an item
*Other item!
[/list]
```

**Minimal reproduction project:**

There's an official RichTextLabel demo that could be used as a playground.

**Comment by bruvzg** [comment]
Lists are implemented in `master` branch, (`ol` and `ul` tags), but without alternative definition variants or separate item tags).

<img width="277" alt="Screenshot 2021-01-25 at 11 09 35" src="https://user-images.githubusercontent.com/7645683/105684586-d77d9300-5efd-11eb-84fb-0f26b733474c.png">

```
Ordered list:
[ol]Item one
[ol type=i]Subitem one
Subitem two
[ol type=a]Subsub item[/ol][/ol]
Item two
[/ol]
Unordered list:
[ul]Item one
Item two
[/ul]
Another variant:
[ol type=A]Item one
Item two
[/ol]
Other list:
[ol type=I]Here's an item
Other item!
[/ol]
```

**Comment by akien-mga** [comment]
Let's close as fixed in 4.0. Thanks!
